aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -192,9 +192,11 @@ struct MeanOps {
 // a set of numbers.
 // `scalar_t` is the type of the input and `acc_t` is the type of the accumulated
 // value. These types differ for complex number input support.
-template <typename scalar_t, typename acc_t=scalar_t>
+template <
+    typename scalar_t,
+    typename acc_t = scalar_t,
+    typename out_t = scalar_t>
```

Comment:
If we're going to have defaults should it not be `out_t = acc_t` for backward compatibility?

Reply:
For the Norm ops, I think it is strange to set `out_t = acc_t`, from the design at https://github.com/pytorch/pytorch/blob/92e03cd583c027a4100a13682cf65771b80569da/aten/src/ATen/native/cpu/Reduce.h#L166-L170, we hope the `project ` return the required output.

Reply:
You have changed the meaning of `NormOps<scalar_t, acc_t>` so it now outputs `scalar_t` instead of `acc_t` which is wrong for complex types where the norm is always real.

Reply:
Thanks for the explanation, changed the default value of `out_t ` to `acc_t `.